sdk/samples/iot/hub/src/paho_iot_hub_methods_example.c

```diff
@@ -301,74 +333,100 @@ static int subscribe()
   return MQTTCLIENT_SUCCESS;
 }
 
-static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message)
+static void receive_messages()
 {
-  (void)context;
   int rc;
+  char* topic = NULL;
+  int topic_len = 0;
+  MQTTClient_message* message = NULL;
+  az_iot_hub_client_method_request method_request;
 
-  printf("Received a message from service.\n");
-  printf("Topic: %s\n", topicName);
+  // Wait until max # messages received or timeout to receive a single message expires.
+  for(uint8_t message_count = 0; message_count < MAX_MESSAGE_COUNT; message_count++)
+  {
+    if (((rc
+          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
+          != MQTTCLIENT_SUCCESS)
+          && (MQTTCLIENT_TOPICNAME_TRUNCATED != rc))
+    {
+      LOG_ERROR("Failed to receive message: MQTTClient return code %d.", rc);
+      exit(rc);
+    }
+    else if (NULL == message)
+    {
+      LOG_ERROR("Timeout expired: MQTTClient return code %d.", rc);
+      exit(rc);
+    }
+    else if (MQTTCLIENT_TOPICNAME_TRUNCATED == rc)
```

Comment:
```suggestion
    else if (rc == MQTTCLIENT_TOPICNAME_TRUNCATED)
```

Reply:
FYI, I have made a note to self to update this in all the samples.